internal/pkg/policy/monitor_integration_test.go

```diff
@@ -128,3 +129,410 @@ func TestMonitor_Integration(t *testing.T) {
 	default:
 	}
 }
+
+func TestMonitor_Debounce_Integration(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	ctx = testlog.SetLogger(t).WithContext(ctx)
+
+	index, bulker := ftesting.SetupCleanIndex(ctx, t, dl.FleetPolicies)
+
+	im, err := monitor.New(index, bulker.Client(), bulker.Client(), monitor.WithDebounceTime(time.Second))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Start index monitor
+	var imerr error
+	var imwg sync.WaitGroup
+	imwg.Add(1)
+	go func() {
+		defer imwg.Done()
+		imerr = im.Run(ctx)
+		if errors.Is(imerr, context.Canceled) {
+			imerr = nil
+		}
+	}()
+
+	m := NewMonitor(bulker, im, config.ServerLimits{})
+	pm, ok := m.(*monitorT)
+	if !ok {
+		t.Fatalf("unable to cast monitor m (type %T) as *monitorT", m)
+	}
+	pm.policiesIndex = index
+
+	var merr error
+	var mwg sync.WaitGroup
+	mwg.Add(1)
+	go func() {
+		defer mwg.Done()
+		merr = m.Run(ctx)
+		if errors.Is(merr, context.Canceled) {
+			merr = nil
+		}
+	}()
+	err = m.(*monitorT).waitStart(ctx)
+	require.NoError(t, err)
+
+	agentID := uuid.Must(uuid.NewV4()).String()
+	policyID := uuid.Must(uuid.NewV4()).String()
+	s, err := m.Subscribe(agentID, policyID, 0, 0)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer m.Unsubscribe(s) //nolint:errcheck // defered function
+
+	policy := model.Policy{
+		PolicyID:       policyID,
+		CoordinatorIdx: 1,
```

Comment:
Could we have a test which tests the coordinator logic too? And adds a new policy revision with `coordinator_idx:0`